ctapipe/reco/shower_max.py

```diff
@@ -77,16 +79,18 @@ def find_shower_max_height(self,energy,h_first_int,gamma_alt):
         c *= np.sin(gamma_alt)
         
         # find the thickness at the height of the first interaction
-        t_first_int = self.interpolate(h_first_int)
+        t_first_int = self._interpolate(h_first_int)
         
         # total thickness at shower maximum = thickness at first interaction + thickness traversed to shower maximum
         t_shower_max = t_first_int + c
         
         # now find the height with the wanted thickness
         for ii, thick1 in enumerate(self.atmosphere.hist):
             if t_shower_max > thick1:
+                
                 height1 = self.atmosphere.bin_lower_edges[0][ii]
                 height2 = self.atmosphere.bin_lower_edges[0][ii-1]
-                thick2  = self.atmosphere.get_value([height2.to(self.atmosphere._binLowerEdges[0].unit).value])[0]
+                thick2  = self.atmosphere.get_value([height2.to(self.atmosphere.bin_lower_edges[0].unit).value])[0]
```

Comment:
_binLowerEdges is the data member of the histogram class (to set up the edges, which is actually not meant to be supported like this, hence the underscore)
bin_lower_edges is a @property and the one that should be used as soon as the binning is properly defined.